mergify_engine/queue/merge_train.py

```diff
@@ -1487,6 +1499,38 @@ async def _split_failed_batches(self, queue_rules: rules.QueueRules) -> None:
                 )
                 break
 
+        # NOTE(sileht): speculative_checks=1 may create car without the
+        # attached draft pull request if this car become the first, it means
+        # the previous car has been merged and we can start testing it
```

Comment:
if spec check is 1, batch size is 2, and in place is allowed, is there a chance that there's no draft PR at all?
Does this work? Tested?

Reply:
yes it works, but I will add a test to prove it!

`TrainCar._create_pull()` is badly named since we introduce in-place, it should be `TrainCar._setup_car()` or `TrainCar._start_checking_car()`.

I will create a ticket for that.